mobile/lib/comparison_report_manager.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:flutter/material.dart';
+import 'package:mobile/app_manager.dart';
+import 'package:mobile/model/gen/anglerslog.pb.dart';
+import 'package:mobile/model/id.dart';
+import 'package:mobile/report_manager.dart';
+import 'package:provider/provider.dart';
+
+class ComparisonReportManager extends ReportManager<ComparisonReport> {
+  static ComparisonReportManager of(BuildContext context) =>
+      Provider.of<AppManager>(context, listen: false)
+          .customComparisonReportManager;
+
+  ComparisonReportManager(AppManager app) : super(app);
+
+  @override
+  ComparisonReport entityFromBytes(List<int> bytes) =>
+      ComparisonReport.fromBuffer(bytes);
+
+  @override
+  Id id(ComparisonReport report) => Id(report.id);
+
+  @override
+  String name(ComparisonReport report) => report.name;
+
+  @override
+  void onDeleteBait(Bait bait) =>
+      entities.removeWhere((id, report) => report.baitIds.contains(bait.id));
```

Comment:
The entities shouldn't be deleted here. The given `Bait` ID should be removed from the entity's `baitIds` property.